File: lib/ciphers/polybius.dart
import 'cipher_base.dart';

class PolybiusCipher implements CipherBase {
  @override
  String get name => "Polybius";
  @override
  String get id => "polybius";

  final Map<String, String> _square = {
    'A': '11', 'B': '12', 'C': '13', 'D': '14', 'E': '15',
    'F': '21', 'G': '22', 'H': '23', 'I': '24', 'K': '24',
    'L': '25', 'M': '31', 'N': '32', 'O': '33', 'P': '34',
    'Q': '35', 'R': '41', 'S': '42', 'T': '43', 'U': '44',
    'V': '45', 'W': '51', 'X': '52', 'Y': '53', 'Z': '54', 'J': '24'
  };

  @override
  String encrypt(String text, String key) {
    return text.toUpperCase().split('').map((char) => _square[char] ?? char).join(' ');
  }
}

File: server/strategies/polybius.py
from strategies.cipher_interface import CipherInterface

class PolybiusCipher(CipherInterface):
    def decrypt(self, text: str, key: str) -> str:
        # Standart Polybius Square (I/J aynı hücrede)
        square = {
            '11': 'A', '12': 'B', '13': 'C', '14': 'D', '15': 'E',
            '21': 'F', '22': 'G', '23': 'H', '24': 'I', '25': 'L',
            '31': 'M', '32': 'N', '33': 'O', '34': 'P', '35': 'Q',
            '41': 'R', '42': 'S', '43': 'T', '44': 'U', '45': 'V',
            '51': 'W', '52': 'X', '53': 'Y', '54': 'Z'
        }
        try:
            # Input format: "23 15 25 25 33" (HELLO)
            parts = text.split()
            return "".join([square.get(p, "?") for p in parts])
        except:
            return "Hata: Geçersiz Polybius formatı."